File: Python/Primer/lists.py
# 2018-06

lst_one = ["remove", "random"]
lst_two = [200, -2, [1.0, 0.0]]

# lists are mutable
lst_one[0] = "ADD"

print(lst_one)
# ['ADD', 'RANDOM']
print(lst_one[1])
# RANDOM

# length
print(len(lst_one))
# 2

# concatenate
lst = lst_one + lst_two
print(lst)
# ['ADD', 'RANDOM', 200, -2, [1.0, 0.0]]

# append
lst.append(None)
print(lst)
# ['ADD', 'random', 200, -2, [1.0, 0.0], None]

# convert to string before sorting
print(sorted(list(map(str,lst))))
# ['-2', '200', 'ADD', 'None', '[1.0, 0.0]', 'random']

# string to list
print(list("100B"))
# ['1', '0', '0', 'B']

File: Scala/Primer/lists.scala
// lists are immutable
val list_one: List[String] = List("Alpha", "Beta")          // List(Alpha, Beta)
val list_two: List[Any] = List(200, -2, List(1.0, 0.0))     // List(200, -2, List(1.0, 0.0))
val list_range: List[Int] = List.range(1, 10)               // List(1, 2, 3, 4, 5, 6, 7, 8, 9)

// concatenate lists
val list_all = list_one ++ list_two                         // List(Alpha, Beta, 200, -2, List(1.0, 0.0))

// append to list by creating a new list
val list_one_new = "Gamma"  :: list_one                     // List(Gamma, Alpha, Beta)

// built-ins
list_one_new.length                                         // 3
list_one_new.sorted                                         // List(Alpha, Beta, Gamma)
list_two.head                                               // 200
list_two.last                                               // List(1.0, 0.0)
list_range.filter(_ > 5)                                    // List(6, 7, 8, 9)

// create list from string
val list_char: List[Char] = ("100B").toList                 // List(1, 0, 0, B)